lib/OpenQA/Schema/Result/Jobs.pm

```diff
@@ -2065,7 +2065,11 @@ sub done ($self, %args) {
     if (($result_unknown || $reason_unknown) && defined $reason) {
         # restart incompletes when the reason matches the configured regex
         my $auto_clone_regex = OpenQA::App->singleton->config->{global}->{auto_clone_regex};
-        $restart = 1 if $result eq INCOMPLETE && $auto_clone_regex && $reason =~ $auto_clone_regex;
+        if ($result eq INCOMPLETE and $auto_clone_regex and $reason =~ $auto_clone_regex) {
+            my $limit = OpenQA::App->singleton->config->{global}{auto_clone_limit};
+            my $ancestors = $self->ancestors($limit + 1);
+            $restart = 1 if $ancestors < $limit;
```

Comment:
please check if each ancestor is *at least* incomplete if not also matching the very same reason for all. I would prefer the latter to be sure we are looking for the very same reason.


```suggestion
            if {$ancestors < $limit) {
                $restart = 1 ;
                $reason = "$reason: auto-cloning based on server setting 'auto_clone_regex'";
            }
            else {
                $reason = "$reason: Not auto-cloning due to server setting 'auto_clone_limit'";
            }
```

Reply:
Wouldn't that make the limit ineffective? Say a job is cloned 19 times and then fails with a different reason.

Maybe this should be considered for a separate change.

Reply:
Well, we could just apply that additional explanation in the case of aborting the restart but IIRC @perlpunk suggested herself to make also the auto-cloning explicit.